python/ray/_private/services.py

```diff
@@ -207,7 +203,14 @@ def find_redis_address(address=None):
     return redis_addresses
 
 
+def find_redis_address_or_die_respect_environment():
```

Comment:
```suggestion
def get_redis_address_to_use_or_die():
   """This infers the redis address to use to connect to Ray, respecting environment overrides."""
```

Reply:
I think this name is _more_ confusing given that I have no idea the difference between `find_redis_address_or_die` and `get_redis_address_to_use_or_die`

Reply:
How about calling it `get_ray_address_to_use_or_die()`. This distinguishes it from the low level redis address finder.

We shouldn't have stuff like `respect_env` since it doesn't make sense unless you know this thing is checking env vars.

Reply:
SGTM